### PR TITLE
Milestone 4 - Category Filter Implementation

### PR DESCRIPTION
**Pull request by acushlakoncept**
In this PR we implemented the category filter function for our Magic-Books Inc application.
- We created the `CHANGE_FILTER` action in actions/index.js
- Also created a filter.js reducer which modifies filter in the Redux store. And defaulted to All.
- And finally rendered the CategoryFilter component in the BooksList component with a conditional method that decides which books are shown in the table depending on the filter value from the Redux store.

### CODE REVIEW

File: src/actions/actionTypes.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// eslint-disable-next-line import/prefer-default-export
```
Comment (bellom) [inline src/actions/actionTypes.js]: **Let's try to look into `eslint` inline comment:**
With the ESLint inline comment, we are indirectly changing the rules by disabling these rules, let's fix the codebase instead.

What could be the reason? If you are using VSCode, you can delete the inline comment and hover over the error(configure eslint checks locally). And you will see your options. Click on the more info button that will take you to a GitHub repo where it is explained why you are getting this error. I find this feature very educational. Please, try to learn and implement the necessary rules so you don't have to keep this disabling line on top of each file. I believe you can do this! 💪😇

File: src/actions/index.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+/* eslint-disable import/prefer-default-export */
```
Comment (bellom) [inline src/actions/index.js]: Also in here, let's fix the codebase instead.

File: src/container/BooksList.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            .map(book => (
+              <Book
+                key={book.id}
+                book={book}
+                handleRemoveBook={() => handleRemoveBook(book.id)}
+              />
+            ))}
+        </tbody>
+      </table>
+      <CategoryFilter />
```
Comment (bellom) [inline src/container/BooksList.js]: Great work guys!

I can see you are rendering `CategoryFilter ` component 👍 , but we also need to implement `handleFilterChange` and pass it down to `CategoryFilter`.

Screenshot from pathwright;
image

File: src/container/BooksList.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+          {allBooks
+            .filter(book => {
+              if (filter !== 'All') {
+                if (book.category === filter) {
+                  return true;
+                }
+                return false;
+              }
+              return true;
+            })
```
Comment (bellom) [inline src/container/BooksList.js]: This is nice 👍 but according to the project requirement, we need to create a function for this.

Let's implement a conditional method to determine the book that would be displayed based on the filter value in our redux store.

Screenshot from pathwright;
image

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi Team :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 

# Status: Changes Required! ♻️ 

**Kindly check out my comments below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Review by MuriloRoque — APPROVED** [review]
## Status: Approved :green_circle: 

### Hello again @acushlakoncept and @ozovalihasan :wave: 

### Thank you for making all of the requested changes. Your project is approved :clap: :clap: :clap: 

### You may merge, good luck with your next projects :rocket: 

Murilo Roque Paiva da Silva

- Github: [@muriloroque](https://github.com/MuriloRoque)
- Linkedin: [MuriloRoque](https://www.linkedin.com/in/murilo-roque-b1268741/)

[Twitter Follow](https://twitter.com/MuriloRoquePai1)

**Pull request merged**